File: Dynamic-Programming/53-Maximum-Subarray/max-subarray.py
class Solution:
    def maxSubArray(self, nums: list[int]) -> int:
        best, current_sum = nums[0], nums[0]        
        for num in nums[1:]:
            current_sum = max(num, current_sum + num)
            best = max(best, current_sum)
        return best

File: Dynamic-Programming/53-Maximum-Subarray/max-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        // applaying Kadane's Algo.
        int len = nums.size();
        int best=nums[0], sum=nums[0];
        for (int k=1; k<len;k++)
        {
            sum = max(nums[k], sum+nums[k]);
            best = max(best, sum);
        }
        return best;
    }
};